src/pivot-table/data/helpers/create-cell.ts

```diff
@@ -83,6 +84,7 @@ const createCell = (
     isEmpty: node.qType === NxDimCellType.NX_DIM_CELL_EMPTY,
     isNull: node.qType === NxDimCellType.NX_DIM_CELL_NULL,
     isPseudoDimension: node.qType === NxDimCellType.NX_DIM_CELL_PSEUDO,
+    isPseudoDimensionBefore: !!(parent?.isPseudoDimensionBefore || parent?.isPseudoDimension),
```

Comment:
This information is kind of already on the cell. You can search up the parents and see if anyone is pseudo, but this skipes that loop

Reply:
Consider renaming it to `isAncestorPseudoDimesion` as that seem to be exactly the kind of value it holds?

Reply:
Ah was looking for the correct term, thanks